### ISSUE TITLE
Why adjusting learning rate is commented out?

### ISSUE THREAD

**Issue by code2graph** [graph4ai/graph4nlp#538]
## ❓ Questions and Help

I see in the NMT example, adjusting learning rate is disabled. 

[self._adjust_lr(epoch)](https://github.com/graph4ai/graph4nlp/blob/master/examples/pytorch/nmt/main.py#L135)
```
        for epoch in range(200):
            self.model.train()
            self.train_epoch(epoch, split="train")

            # self._adjust_lr(epoch)
            if epoch >= 0:
```

I am wondering whether that was done purposefully? You were getting bad result when it was enabled? 

Would appreciate if you share your insight.

**Comment by AlanSwift** [comment]
No, I just comment on it during debugging since I didn't tune it.

**Comment by code2graph** [comment]
@AlanSwift what I have to do to tune it? Not sure whether I understood it. It felt to me the same adjusting learning rate was used for other applications as well. My question is:

1. What do you mean by you have not tune it?
2. I would be happy to follow the steps if you stare with me and I can contribute back to the library.

**Comment by AlanSwift** [comment]
The early stop is a good trick to get better performance but you have to tune the stop time. But I haven't tune it.

**Comment by code2graph** [comment]
@AlanSwift can you please give me some pointers how can I tune it? I would be happy to work on it and contribute a PR.

**Comment by code2graph** [comment]
@AlanSwift can you please provide any pointers 🙏

**Comment by code2graph** [comment]
@AlanSwift sorry for pinging you again with this issue. 

Can you please provide me with some pointers and I am happy to work that, tune it and then contribute back to the library. Would really appreciate your help.

**Comment by AlanSwift** [comment]
I'm really sorry for the late reply since I'm busy the past few days.
You can uncomment this line and tune these parameters: lr_start_decay_epoch, lr_decay_per_epoch, lr_decay_rate.
The current lr schedule implementation is ExponentialLR.
I suggest you use the official PyTorch implementation: [ExponentialLR](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.ExponentialLR.html). And there are some [other learning rate schedule methods](https://pytorch.org/docs/stable/search.html?q=torch.optim.lr_scheduler&check_keywords=yes&area=default) if you are interested.
